scripts/zones/Ceizak_Battlegrounds/npcs/qm_mastop.lua

```diff
@@ -0,0 +1,29 @@
+-----------------------------------
+-- Zone: Ceizak Battlegrounds
+-- NPC: qm_mastop (???)
+-- Spawns Mastop
+-- !pos -230 0 36 261
+-----------------------------------
+local ID = zones[xi.zone.CEIZAK_BATTLEGROUNDS]
+-----------------------------------
+local entity = {}
+
+entity.onTrade = function(player, npc, trade)
+    if
+        npcUtil.tradeHas(trade, xi.item.CELADON_YGGRETE_SHARD_IV) and
```

Comment:
We may want to use `npcUtil.tradeHasExactly()` so players dont put other items in there and get screwed.